recipe/meta.yaml

```diff
@@ -71,7 +71,7 @@ outputs:
         - run_cmake_cuda_tests.sh  # [linux]
       commands:
         - nvcc --version
-        - $CXX --verbose ${CXXFLAGS} -lcuda -lcudart_static test.cpp
+        - $CXX --verbose ${CXXFLAGS} ${LDFLAGS} -lcuda -lcudart_static test.cpp
```

Comment:
Not sure about this. Thought `LDFLAGS` should go to the linker. Not the compiler

Reply:
Whomever wrote this test is using $CXX as both the linker and the compiler. GCC will pass linker flags through to the linker (probably ld in this case) for convenience. This could be two commands: one for compiling and a separate for linking. In that case, LDFLAGS would only be needed for the linking step.

Reply:
That might have been me. Sure we can restructure as you have suggested

Reply:
So, it seems that some package is adding non-ld arguments to LDFLAGS, so this type of separate compiling and linking doesn't work. Specifically, LDFLAGS contains '-Wl,-O2' which are not linking arguments (maybe others too?).

```bash
+ ld --verbose test.o -o test -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,--allow-shlib-undefined -Wl,-rpath,/home/conda/feedstock_root/build_artifacts/cuda-nvcc-split_1685985242581/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib -Wl,-rpath-link,/home/conda/feedstock_root/build_artifacts/cuda-nvcc-split_1685985242581/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib -L/home/conda/feedstock_root/build_artifacts/cuda-nvcc-split_1685985242581/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib -L/home/conda/feedstock_root/build_artifacts/cuda-nvcc-split_1685985242581/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/targets/x86_64-linux/lib -L/home/conda/feedstock_root/build_artifacts/cuda-nvcc-split_1685985242581/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/targets/x86_64-linux/lib/stubs -L/home/conda/feedstock_root/build_artifacts/cuda-nvcc-split_1685985242581/_build_env/targets/x86_64-linux/lib -L/home/conda/feedstock_root/build_artifacts/cuda-nvcc-split_1685985242581/_build_env/targets/x86_64-linux/lib/stubs -lcuda -lcudart_static
ld: unrecognized option '-Wl,-O2'
ld: use the --help option for usage information
GNU ld version 2.27-44.base.el7_9.1
  Supported emulations:
   elf_x86_64
   elf32_x86_64
   elf_i386
   elf_iamcu
   i386linux
   elf_l1om
   elf_k1om
Tests failed for cuda-nvcc_linux-64-12.0.76-hba56722_6.conda - moving package to /home/conda/feedstock_root/build_artifacts/broken
```